matchzoo/generators/pair_generator.py

```diff
@@ -0,0 +1,128 @@
+"""Matchzoo pair generator."""
+
+from matchzoo import engine
+from matchzoo import datapack
+from matchzoo import utils
+
+import pandas as pd
+import numpy as np
+import typing
+
+
+class PairGenerator(engine.BaseGenerator):
+    """PairGenerator for Matchzoo.
+
+    Pair generator can be used only for ranking.
+
+    Examples:
+        >>> relation = [['qid0', 'did0', 0],
+        ...             ['qid0', 'did1', 1],
+        ...             ['qid0', 'did2', 2]
+        ...            ]
+        >>> columns = ['id_left', 'id_right', 'label']
+        >>> content = {'qid0': [1, 2],
+        ...            'did0': [2, 3],
+        ...            'did1': [3, 4],
+        ...            'did2': [4, 5],
+        ...           }
+        >>> input = datapack.DataPack(relation=relation,
+        ...                           content=content,
+        ...                           columns=columns
+        ...                           )
+        >>> from matchzoo.generators import PairGenerator
+        >>> generator = PairGenerator(input, 1, 1, 1, 'train', True)
+        >>> assert generator
+        >>> len(generator)
+        2
+        >>> x, y = generator[0]
+        >>> assert x is not None
+        >>> assert y is not None
+
+    """
+
+    def __init__(
+        self,
+        inputs: datapack.DataPack,
+        num_neg: int=1,
+        dup_time: int=4,
```

Comment:
maybe it's better to change the variable name, such as `num_dup`, use the same naming convention as `num_neg`